### PR TITLE
ci: bump ci image to rust 1.73.0

### PR DESCRIPTION
**Pull request by altaua**

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3911545

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3912962

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3913385

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3913484

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3913631

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3913630

**Comment by altaua** [comment]
bot update-ui latest -v CMD_IMAGE=paritytech/ci-unified:bullseye-1.73.0-2023-05-23-v20231009

**Comment by altaua** [comment]
bot update-ui default -v CMD_IMAGE=paritytech/ci-unified:bullseye-1.73.0-2023-05-23-v20231009

**Comment by altaua** [comment]
bot update-ui -v CMD_IMAGE=paritytech/ci-unified:bullseye-1.73.0-2023-05-23-v20231009

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3929065

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3929203

**Comment by altaua** [comment]
bot fmt

**Review by alvicsam — APPROVED** [review]

**Review by sergejparity — APPROVED** [review]

File: substrate/client/network/src/protocol/notifications/behaviour.rs [hunk 1]
```diff
@@ -1423,7 +1423,6 @@ impl NetworkBehaviour for Notifications {
 									let delay_id = self.next_delay_id;
 									self.next_delay_id.0 += 1;
 									let delay = futures_timer::Delay::new(ban_duration);
-									let peer_id = peer_id;
```
Comment (kianenigma) [inline substrate/client/network/src/protocol/notifications/behaviour.rs:1426]: wonder why these were here before?

**Review by kianenigma — APPROVED** [review]

File: substrate/client/network/src/protocol/notifications/behaviour.rs (hunk 1 above)
Comment (altaua) [reply substrate/client/network/src/protocol/notifications/behaviour.rs:1426]: it used to be `let peer_id = *peer_id;` until [this commit](https://github.com/paritytech/substrate/commit/4617267aa82f), which just removed the `*` and left the remaining no-op line around.

**Review by liamaharon — APPROVED** [review]

File: .cargo/config.toml [hunk 2]
```diff
@@ -30,4 +30,6 @@ rustflags = [
   "-Aclippy::derivable_impls",           # false positives
   "-Aclippy::stable_sort_primitive",     # prefer stable sort
   "-Aclippy::extra-unused-type-parameters", # stylistic
+  "-Aclippy::default_constructed_unit_structs", # stylistic
+  "-Aclippy::bad-bit-mask",              # triggered by macro behaviour in bitflags 1.3.2
```
Comment (ggwpez) [inline .cargo/config.toml:34]: Can we disable this one specifically in the files that need it?

Comment (altaua) [reply .cargo/config.toml:34]: ohh right I forgot you can disable lints at a file level too, will do. (normal line-level lint bypasses ofc don't work on macro-generated code)

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3938419

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3938451

**Comment by ggwpez** [comment]
Can you also check if its worth to update the Rust version that the `bot fmt` uses please?  
Currently we are on `nightly-2023-05-16`.

**Comment by altaua** [comment]
> Can you also check if its worth to update the Rust version that the `bot fmt` uses please? Currently we are on `nightly-2023-05-16`.

yeah that's on the todo list too, nightly bumps are always a huge hassle though

**Pull request merged**